src/components/FileBrowser.vue

```diff
@@ -0,0 +1,279 @@
+    },
+    changeLocation(item) {
+      const { type, id, name } = item;
+      if ((this.location.id !== id || this.location.type !== type)
+          && type !== 'item') {
+        this.$emit('update:location', { type, id, name });
+      }
+    },
+    refresh() {
+      this.refreshCounter_ += 1;
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+v-data-table.girder-file-browser-component.elevation-1(
+    select-all,
+    :headers-length="4",
+    v-model="selected",
+    :pagination.sync="pagination",
+    :items="rows",
+    :total-items="totalItems",
+    :loading="loading ? 'accent' : false",
+    item-key="id")
+
+  //- Header Slot
+  template(slot="headers", slot-scope="props")
+    tr.secondary.lighten-5
+      th(width="1%")
+        v-checkbox.secondary--text.text--darken-1.pr-2(
+            color="accent",
+            hide-details,
+            :input-value="props.all",
+            :indeterminate="selected.length > 0 && !props.all",
+            @click.native="toggleAll")
+      th.pl-0(width="80%")
+        v-breadcrumbs.pl-0
+          v-icon.mdi-24px(slot="divider", color="accent") {{ $vuetify.icons.chevron }}
+          v-breadcrumbs-item(@click.native="changeLocation(breadcrumb.root)")
+            v-icon.mdi-24px(color="accent") {{ $vuetify.icons.globe }}
+            | &nbsp; {{ breadcrumb.root.name }}
+          v-breadcrumbs-item(
+              v-for="item in breadcrumb.path",
+              :key="`${item.id}.crumb`",
+              @click.native="changeLocation(item)") {{ item.name }}
+      th.pr-0(width="1%")
+        v-btn(flat,
+            small,
+            color="secondary darken-2",
+            v-if="newFolderEnabled",
+            @click="$emit('click:newfolder')")
+          v-icon.mdi-24px.mr-1(left, color="accent") {{  $vuetify.icons.folderNew }}
+          | New Folder
+      th.pl-0.pr-2(width="1%")
+        v-btn(flat,
+            small,
+            color="secondary darken-2",
+            v-if="newItemEnabled",
+            @click="$emit('click:newitem')")
+          v-icon.mdi-24px.mr-1(left, color="accent") {{  $vuetify.icons.fileNew }}
+          | New Item
+
+  //- Table Row Slot
+  template(slot="items", slot-scope="props")
+    tr.itemRow(:active="props.selected",
+        @click="props.selected = !props.selected",
+        :key="props.index")
+      td
+        v-checkbox.secondary--text.text--darken-1.pr-2(
+            :input-value="props.selected", accent, hide-details)
+      td.pl-1
+        span.text-container.secondary--text.text--darken-3(
+            :class="{selectable: props.item.type !== 'item'}",
+            @click.stop="changeLocation(props.item)")
+          v-icon(:color="props.selected ? 'accent' : ''") {{ $vuetify.icons[props.item.icon] }}
+          | &nbsp; {{ props.item.name }}
+      td.text-xs-right.secondary--text.text--darken-3(colspan="2") {{ props.item.size }}
+</template>
+
+<style lang="stylus" scoped>
+@import '~vuetify/src/stylus/settings/_colors.styl'
```

Comment:
Hi @subdavis, I just noticed that you are importing Vuetify stylus in this PR. But the recent master change to SCSS will prevent you from doing this after rebasing master. But at the same time, I think switching to JavaScript color is better because using JavaScript color will make it adaptive to individual app theme and, as Vuetify document states, using stylus color will increase CSS export by~30kb.